### PR TITLE
[ORT Training] Some important updates of ONNX Runtime training APIs

### LINKED ISSUE #1327: AttributeError: Can't pickle local object 'main.<locals>.train_transforms'
### System Info

```shell
python==3.8

accelerate==0.22.0
evaluate==0.4.0
optimum==1.12.0
scikit-learn==1.3.0
transformers==4.32.1

onnx==1.14.0
onnxruntime-training==1.15.0+cu118
```

### Who can help?

@JingyaHuang @echarlai

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

Environment:
```Dockerfile
FROM mcr.microsoft.com/aifx/acpt/stable-ubuntu2004-cu118-py38-torch200
RUN pip install accelerate evaluate optimum transformers
RUN pip install scikit-learn

RUN pip list
```

Run Command:
```bash
torchrun run_image_classification.py \
                        --model_name_or_path google/vit-base-patch16-224 \
                        --do_train --do_eval \
                        --dataset_name beans \
                        --fp16 True --num_train_epochs 1 \
                        --per_device_train_batch_size 8 --per_device_eval_batch_size 8 \
                        --remove_unused_columns False --ignore_mismatched_sizes True \
                        --output_dir output_dir --overwrite_output_dir \
                        --dataloader_num_workers 1
```

Error:
```
Traceback (most recent call last):
  File "run_image_classification_ort.py", line 423, in <module>
    main()
  File "run_image_classification_ort.py", line 397, in main
    train_result = trainer.train(resume_from_checkpoint=checkpoint)
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/optimum/onnxruntime/trainer.py", line 462, in train
    return inner_training_loop(
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/optimum/onnxruntime/trainer.py", line 739, in _inner_training_loop
    for step, inputs in enumerate(train_dataloader):
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/accelerate/data_loader.py", line 381, in __iter__
    dataloader_iter = super().__iter__()
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 442, in __iter__
    return self._get_iterator()
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 388, in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 1043, in __init__
    w.start()
  File "/opt/conda/envs/ptca/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/opt/conda/envs/ptca/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/opt/conda/envs/ptca/lib/python3.8/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/opt/conda/envs/ptca/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/opt/conda/envs/ptca/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/opt/conda/envs/ptca/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/opt/conda/envs/ptca/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'main.<locals>.train_transforms'
```

### Expected behavior

The script is expected to train `google/vit-base-patch16-224` for image-classification task.

The error occurs only when using `--dataloader_num_workers` parameter. The error callstack never passes through any onnxruntime code and seems to be isolated to the dataloader. I believe some update to the accelerate package is causing this issue with optimum. There is no error when I use `ORTModule` directly from `onnxruntime.training.ortmodule` library, only an issue when I use `ORTTrainer` from `optimum.onnxruntime` library.

Related links from the web:
- https://discuss.pytorch.org/t/cant-pickle-local-object-dataloader-init-locals-lambda/31857/9
- https://stackoverflow.com/questions/62186218/python-multiprocessing-attributeerror-cant-pickle-local-object

This is URGENT as the AzureML Vision Team is in the final steps of releasing finetuning components for image-classification to production. Thank you in advance for any help!

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

- [x] Update `ORTTrainer` to be compatible with transformers [40ea9ab2a1ad99f12e71c3a26215ad33df082ef9](https://github.com/huggingface/transformers/pull/26751).
- [x] /*Deprecation*/ Deprecate the evaluation and prediction using ONNX runtime
- [x] Refactoring of examples, tests and enable again the CI -> Run with tiny functional models

[Deprecation notes]

The Optimum team decided to deprecate the evaluation and prediction using ONNX Runtime for the reasons below:
* After the deprecation, evaluation and prediction for the trained model are always possible within PyTorch through `ORTTrainer`. If you want to do inference with ORT, either the evaluation or the prediction can be done through ORTModels in the library.
* Reduce the workload of maintaining ORT inference, which could be broken easily with the evolution of ORT inference APIs. And the feature has no clear usage to encourage us to continue maintaining it.
* Ease the maintenance of `ORTTrainer` and training examples

[Other subjects to discuss]

* Automate the update and tests of ORT training examples

### CODE REVIEW

File: optimum/onnxruntime/trainer_seq2seq.py [hunk 1]
```diff
@@ -48,6 +49,9 @@
 from .utils import ONNX_DECODER_NAME, ONNX_DECODER_WITH_PAST_NAME, ONNX_ENCODER_NAME, wrap_onnx_config_for_loss
 
 
+if is_accelerate_available():
+    from accelerate import __version__ as accelerate_version
```
Comment (fxmarty) [inline optimum/onnxruntime/trainer_seq2seq.py:53]:
```suggestion
if is_accelerate_available():
    from accelerate import __version__ as accelerate_version
else:
    raise ImportError(
        "The package `accelerate` is required to use the ORTTrainer. Please install it following https://huggingface.co/docs/accelerate/basic_tutorials/install."
    )
```

**Comment by fxmarty** [comment]
https://github.com/huggingface/optimum/issues/1327 should already be fixed

**Comment by JingyaHuang** [comment]
yeah @fxmarty I forgot to update the description, thanks a lot for the fix.

**Review by fxmarty — APPROVED** [review]
LGTM, hopefully adding `position_ids` inputs did not break ORTTrainer?

**Review by regisss — APPROVED** [review]
LGTM!

**Comment by JingyaHuang** [comment]
All tests passed:
```
====================================================== test session starts =======================================================
platform linux -- Python 3.10.0, pytest-7.4.2, pluggy-1.0.0 -- /home/onnxruntimedev/miniconda3/bin/python
cachedir: .pytest_cache
rootdir: /workspace
configfile: pyproject.toml
collected 13 items                                                                                                               

onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp16_0_distilbert_text_classification PASSED  [  7%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp16_1_gpt2_text_generation PASSED            [ 15%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp16_2_t5_text2text_generation PASSED         [ 23%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp32_0_distilbert_text_classification PASSED  [ 30%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp32_1_gpt2_text_generation PASSED            [ 38%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp32_2_t5_text2text_generation PASSED         [ 46%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp32_with_label_smoothing_0_distilbert_text_classification PASSED [ 53%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp32_with_label_smoothing_1_gpt2_text_generation PASSED [ 61%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationTest::test_trainer_fp32_with_label_smoothing_2_t5_text2text_generation PASSED [ 69%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationDeepSpeedTest::test_trainer_fp16_ds_stage1_0_distilbert_text_classification PASSED [ 76%]
onnxruntime/nightly_test_trainer.py::ORTTrainerIntegrationDeepSpeedTest::test_trainer_fp16_ds_stage2_0_distilbert_text_classification PASSED [ 84%]
onnxruntime/nightly_test_trainer.py::ORTTrainerOptimizerChoiceTest::test_ort_fused_adam PASSED                             [ 92%]
onnxruntime/nightly_test_trainer.py::ORTTrainerExampleTest::test_trainer_glue SKIPPED (skip for now, server socket error)  [100%]
```

Left an example test skipped, need to add similar tests for other tasks.

**Pull request merged**